modules/islandora_rest_api_auth/islandora_rest_api_auth.module

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/**
+ * @file
+ * Module hooks for the Islandora REST API Auth module.
+ */
+
+/**
+ * Implements hook_init().
+ */
+function islandora_rest_api_auth_init() {
+  islandora_rest_api_auth_authenticate();
+}
+
+/**
+ * Implements hook_form__FORMID_alter().
+ */
+function islandora_rest_api_auth_form_user_profile_form_alter(&$form, &$form_state, $form_id) {
+  global $user;
+
+  module_load_include('inc', 'islandora_rest', 'includes/utilities');
+  $access = FALSE;
+  foreach (user_role_permissions($user->roles) as $perms) {
```

Comment:
Just because we did have a quick discussion about it: We have the potential for keys to exist in our table which the user is no longer allowed to use, either due to their assigned roles or the permissions thereby granted changing... let's ignore it for now... Shouldn't really break anything... sure it'll allow the user swapping business, but the normal access callbacks should then immediately deny access to whatever action was attempted.